### DISCUSSION TITLE
What would you like to see in video tutorials?

### DISCUSSION THREAD

**Discussion by elementbound** [foxssake/netfox#201] [Announcements]
Hi All,

I'm considering making some video tutorials, for the eventual 2.x line, to give the API some time to settle.

With that, if there's demand / support from you all, I'd make some videos along with the documentation. My question is, what would you like to see in these videos? What kind of tutorials would help you in building your games with netfox? Any areas you'd like more clarity on? Go wild, I'm all ears for ideas :) 

Note that I'm personally a fan of short, to the point videos, so I'd probably do something similar with the videos.

Also note that this post is **not a promise.** Whether or not this becomes a thing depends on community demand - I'd like to build things that help you, and not put effort time into things that don't :)

**Comment by ghost** [comment]
I really like Nakama tutorials, that uses generic daily examples and are quick.

Also I would love to see some tutorials on syncing projectiles/non-player bodies.

**Comment by Andicraft** [comment]
One thing I'd love to see is hitscan weapons and how to get those working.

Like a full tutorial series building the basics of a Quake 3 style FPS would be dope, but that's a lot to ask for, so I'd love to see that slice of it 😃

**Reply by elementbound** [reply]
Hitscan weapons with rollback could be a stretch due to how Godot handles physics :grimacing: But you're definitely not the first one to mention this, so I definitely need to look into it! 

Also, building a multiplayer game in a video series would make a lot of sense to me, we'll just have to see what kind of game(s)!

**Comment by elementbound** [comment]
@TheYellowArchitect you've mentioned video tutorials earlier, I'd love to have your thoughts in this thread!

**Reply by TheYellowArchitect** [reply]
Sorry for the late reply, was without internet till the last spam on issues :sweat_smile: 
Video tutorials to expand on forest-brawl (e.g. new skills) would be good, but I think the most important is how to reach forest-brawl level from having exclusively netfox and netfox.extras code.
**Essentials**: How to spawn a player, move the player, and have him spawn a projectile which damages other players.

Theoritically how to remake forest-brawl hahaha

Adding stuff to the custom initial game state on forest-brawl was very hard: In my usage of netfox, I found that the existing spawn points were client-based. So when a player spawns, the server does not determine *where* that player spawns. I wished to inject this "**the server** dictates where to spawn" (see `snap_to_spawn()` function) into the **existing** logic which spawns players at a predetermined location, and the result while succesful, was unencouraging. So the "initialize game state" part should be included for sure. To tweak the existing forest brawl's example, or have the dev make his own.

Lastly, the dev will certainly want to add more things in his game (e.g. items), so **explaining how to add new properties** (e.g. item1) to sync must be included towards the end of the series.

**Video-editor pro-tip**: Start the series with an "This is the end-result of our tutorial" or sth, demonstrating instantly to the viewer what the end-goal is. Show forest-brawl for example at 150 ms, and highlight rollbacks.
Basically you show the "target result" forest-brawl in-game, at the very start, while explaining rollback (the smooth introduction) and that ofc its server-client (redirect to snopek if P2P xd)
Around here show how to install it via asset library (fast-forward ofc but there are many who dont know), or via github (download and drag and drop lol)

Somewhere mention that the physics accuracy don't really matter, since the server's physics is always the authority. Seems minor but for me this saved me from searching autistically accurate physics addons, and trying to meld them into my game.

I don't have anything else in my mind atm, will write more if I think of something important
